### DISCUSSION TITLE
Handling DeleteAation's foreign key constraints

### DISCUSSION THREAD

**Discussion by sergejostir** [filamentphp/filament#8869] [Help] [answered]
Hello.

How to efficiently handle the foreign key violations in Filament? The most DRY option would be to use Laravel exception handler, but I can't figure out how to then generate a response that would halt or cancel the Filament action.

I would really like not to use "before" method on every single affected DeleteAction.

Thanks

**Comment by VincvanAs** [comment]
You could create an observer or use the boot method in a model, you could use something like this but then for deleting:

```
protected static function boot(): void
    {
        parent::boot();

        static::updating(function ($category) {
            if ($category->isDirty('active') && !$category->active && $category->articles()->exists()) {
                Notification::make()
                    ->title(__('category.error'))
                    ->body(__('category.active_remains_error'))
                    ->danger()
                    ->send();
                return false;
            }
        });
    }

```

**Reply by sergejostir** [reply]
But then I have to manually specify relations. I would like to depend on QueryException that is thrown because of MySQL's 1451 error (foreign constraint violation).

**Comment by sergejostir** [comment, marked as answer]
Even though it's not documented, the DeleteAction also uses CanCustomizeProcess trait which allows you to customize the process. I implemented the exception handler with the "using" method.
